ideas/150-gas-abstraction.md

```diff
@@ -0,0 +1,92 @@
+# Gas Abstraction
+
+## Preamble
+
+    Idea: 150-gas-abstraction
+    Title: Gas Abstraction
+    Status: In Progress
+    Created: 2018-02-01
+    Requires: 145-identity
+    Replaces: 073-snt-as-gas
+
+## Summary
+
+Enable Status users to pay gas fee with any token that is valueable to society. 
+Users would sign a message request to contracts which will use user's Identity balance of selected token to refund proportional used gas to a incentivezed ETH owner to include that message in a transction.
+
+## Swarm Participants
+
+- Lead Contributor: Ricardo Guilherme Schmidt
+- Contributor: Richard Ramos
+- Contributor: Iuri Matias
+- UX (Mist): Alexandre Van de Sande 
+- UX (Status): 
+
+
+## Product Overview
+
+A barrier for user adoption to blockchain through Status would be the need of holding two tokens for paying fees, ether gas for moving the [SNT](https://etherscan.io/address/0x744d70fdbe2ba4cf95131626614a1763df805b9e#readContract) fee from user balance to decentralized service provider. 
+
+The product solve this by emulating a gas abstraction by adding "Gas Relayer Actor" in top of smart contracts as: [Identity](https://github.com/status-im/contracts/blob/73-economic-abstraction/contracts/identity/Identity.sol)[GasRelay](https://github.com/status-im/contracts/blob/73-economic-abstraction/contracts/identity/IdentityGasRelay.sol), MultiSigWallet, [SNTController](https://github.com/status-im/contracts/blob/73-economic-abstraction/contracts/status/SNTController.sol#L82) (updated from SNTPlaceHolder). 
+
+- Allows SNT holder to broadcast ethereum signed messages to anyone with ETH to validate them in the GasRelay smart contracts, and offering a refund of gas used in a gas price set in SNT. 
+- Whoever have ether could verify if the gas price worth the SNT gas price offered.
+- This becomes a micro trade of ETH->SNT.
+- Allow user to use any token they want (even ETH, or ETH stored in Identity).
+
+### Product Description
+
+Gas Relay node: Status Destkop extension or an independent node can include messages by making automatic transactions calls to smart contracts to earn tokens being offered as gas price refund.
```

Comment:
I don't understand these things, can you elaborate or link to elaboration? Gas relay node / adaptors.

Reply:
Gas relay node This are nodes that will be listening to messages in swarm, this messages would give them contract defined "token gas" to who includes it in a transaction.

Adaptor is the function in contract that processes the ethereum signed message, allows the call by gas relay node, and pay the gas relay node the gas in tokens controlled by this contract.